### DISCUSSION TITLE
Can GCNConv layer handle adjacency matrices of weighted directed graphs that have negative weights?

### DISCUSSION THREAD

**Discussion by Laknath1996** [pyg-team/pytorch_geometric#2695] [Q&A]
I am dealing with a dataset of weighted directed graphs and some of their adjacency matrices have negative entries. I am hoping to construct a simple GCN-based network for them, but when I attempt to train the network, the loss is "nan" from the very beginning. Therefore, I was wondering whether the adjacency matrices supported by the GCNConv layer allow negative entries within them. I'd be grateful if you could clarify my doubt.

**Comment by rusty1s** [comment]
Please seem my answer here: https://github.com/rusty1s/pytorch_geometric/issues/397#issuecomment-855639117
